src/app/cli/src/coda_run.ml

```diff
@@ -83,6 +86,113 @@ let log_shutdown ~conf_dir ~top_logger coda_ref =
+      ~f:(fun t ->
+        Coda_commands.get_status ~flag:`Performance t
+        |> Daemon_rpcs.Types.Status.to_yojson )
+  in
+  Yojson.Safe.to_file status_file status ;
+  (*coda logs*)
+  let coda_log = conf_dir ^/ "coda.log" in
+  let () =
+    match Core.Sys.file_exists coda_log with
+    | `Yes ->
+        let coda_short_log = temp_config ^/ "coda_short.log" in
+        (*get the last 4MB of the log*)
+        let log_size = 4 * 1024 * 1024 |> Int64.of_int in
+        let log =
+          In_channel.with_file coda_log ~f:(fun in_chan ->
+              let len = In_channel.length in_chan in
+              In_channel.seek in_chan
+                Int64.(max 0L (Int64.( + ) len (Int64.neg log_size))) ;
+              In_channel.input_all in_chan )
+        in
+        Out_channel.write_all coda_short_log ~data:log
+    | _ ->
+        ()
+  in
+  (*System info/crash summary*)
+  let uname = Core.Unix.uname () in
+  let daemon_command = sprintf !"Command: %{sexp: string array}" Sys.argv in
+  let sys_info : Yojson.json =
+    `Assoc
+      [ ("OS_type", `String Sys.os_type)
+      ; ("Release", `String (Core.Unix.Utsname.release uname))
+      ; ("Machine", `String (Core.Unix.Utsname.machine uname))
+      ; ("Sys_name", `String (Core.Unix.Utsname.sysname uname))
+      ; ("Crash_time", `String (Time.to_string crash_time))
+      ; ("Exception", `String exn_str)
+      ; ("Command", `String daemon_command) ]
+  in
+  Yojson.to_file (temp_config ^/ "crash_summary.json") sys_info ;
+  (*copy daemon_json to the temp dir *)
+  let daemon_config = conf_dir ^/ "daemon.json" in
+  let _ =
+    match Core.Sys.file_exists daemon_config with
+    | `Yes ->
+        Core.Sys.command
+          (sprintf "cp %s %s" daemon_config (temp_config ^/ "daemon.json"))
+    | _ ->
+        0
+  in
+  (*Zip them all up*)
+  let tmp_files =
+    [ "coda_short.log"
+    ; "registered_mask.dot"
+    ; "frontier.dot"
+    ; "coda_status.json"
+    ; "crash_summary.json"
+    ; "daemon.json" ]
+  in
+  let files = tmp_files |> String.concat ~sep:" " in
+  let tar_command =
+    sprintf "tar  -C %s -cJf %s %s" temp_config report_file files
+  in
+  let exit = Core.Sys.command tar_command in
+  let action_string =
+    if exit = 2 then (
+      Core.eprintf !"Error making the crash report. Exit code: %d\n" exit ;
+      sprintf
+        "include the last 20 lines from .coda-config/coda.log and then paste \
+         the following:\n\
+         Status:\n\
+         %s\n\
+         Exception:\n\
+         %s\n"
+        (Yojson.Safe.to_string status)
+        exn_str )
+    else sprintf "attach the crash report %s" report_file
+  in
+  let _ = Core.Sys.command (sprintf "rm -rf %s" temp_config) in
+  Core.eprintf
+    !{err|
+
+  ☠  Coda Daemon crashed. The Coda Protocol developers would like to know why!
+
+  Please:
+    Open an issue:
+      https://github.com/CodaProtocol/coda/issues/new
+
+    Briefly describe what you were doing and %s
```

Comment:
Can you update this message to ask people to upload the zip crash report?

Reply:
It already does? `action_string` refers to that message